src/views/pages/page.njk

```diff
@@ -2,40 +2,71 @@
 {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
 {% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
 {% from "govuk/components/radios/macro.njk" import govukRadios %}
-{% from "pages/components/macro.njk" import multiSideBar,
-conditionalButtons %}
+{%
+  from "pages/components/macro.njk" import multiSideBar,
+  conditionalButtons
+%}
 {% set items = meta.items %}
+{% set errors = meta.errors %}
 {% set sidebar = meta.sidebar %}
 
 {% block pageTitle %}
-  {% if errorList %}
+
+  {% if errors[meta.currentPageId] %}
     Error:
   {% endif %}
   {{ pageTitle }}
 {% endblock %}
 
 {% block beforeContent %}
   {% if meta.previousPageId %}
-    {{ govukBackLink({
-      id: "backLink",
-      text: "Back",
-      href: meta.previousPageId
-    }) }}
-  {% endif %}
-  {% if errorList %}
-    {{ govukErrorSummary ({
-        "titleText": "There is a problem",
-        "errorList": errorList
-      }) }}
+    {{
+      govukBackLink({
+        id: "backLink",
+        text: "Back",
+        href: meta.previousPageId
+      })
+    }}
   {% endif %}
+
+  
 {% endblock %}
 
 {% block content %}
   <div id="content" class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds govuk-!-margin-top-2">
+      {% if errors[meta.currentPageId] %}
+        {{
+          govukErrorSummary ({
```

Comment:
Just a thought but would it be cleaner by defining a helper block or macro for the error handling:

```
{% macro withError(config, errorMessage) %}
  {% if errorMessage %}
    {% set config = config | merge({errorMessage: {text: errorMessage}}) %}
  {% endif %}
  {{ govukRadios(config) }}
{% endmacro %}
```

and then calling this macro in the form:

```
{% if meta.questionType === 'radio' %}
  {% set radioConfig = items %}
  {{ withError(radioConfig, errors[meta.currentPageId]?.message) }}
{% endif %}
```

Just a thought as it would improve readability. WDYT?

Reply:
As we are unknown whether we will continue, I think leaving is fine. Can be abstracted out if we ever do another question page.

Reply:
Same as below.

Reply:
Let's drop this and come back to it if we have a potential reuse on the horizon. For now this is only used in the page.njk.